### PR TITLE
UCT/IB: Fix create STRICT_ORDER key

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
@Artemy-Mellanox pls port this to v1.9.x

File: test/gtest/uct/ib/test_ib_md.cc [hunk 1]
```diff
@@ -111,17 +114,22 @@ bool test_ib_md::has_ksm() const {
 }
 
 bool test_ib_md::check_umr(uct_ib_md_t *ib_md) const {
+    if (ib_md->dev.flags & UCT_IB_DEVICE_FLAG_MLX5_PRM) {
 #if HAVE_DEVX
-    return has_ksm();
+        if (ucs_derived_of(ib_md, uct_ib_mlx5_md_t)->flags & UCT_IB_MLX5_MD_FLAG_DEVX) {
+            return has_ksm();
+        } else {
+            return ib_md->relaxed_order;
```
Comment (yosefe) [inline test/gtest/uct/ib/test_ib_md.cc:122]: BTW, if we have relaxed_order, shouldn't we always have UMR, regardless of devx/exp situation?
for example even if has_ksm() would return false

Comment (Artemy-Mellanox) [reply test/gtest/uct/ib/test_ib_md.cc:122]: we should, but in current implementation DEVX && !KSM is exception

Comment (yosefe) [reply test/gtest/uct/ib/test_ib_md.cc:122]: i see, so this ok for v1.9 branch, but can you pls adjust this test here to have the more generic logic?

Comment (Artemy-Mellanox) [reply test/gtest/uct/ib/test_ib_md.cc:122]: you mean modify implementation for old FW?

Comment (yosefe) [reply test/gtest/uct/ib/test_ib_md.cc:122]: i mean make the test more independent of ib init logic
e.g if we have relaxed_order - we always expect 2nd key - so always return true

Comment (Artemy-Mellanox) [reply test/gtest/uct/ib/test_ib_md.cc:122]: but the test will fail than

File: test/gtest/uct/ib/test_ib_md.cc [hunk 2]
```diff
@@ -75,11 +75,14 @@ void test_ib_md::ib_md_umr_check(void *rkey_buffer,
...
 
+    status = uct_md_mkey_pack(md(), memh, rkey_buffer);
+    EXPECT_UCS_OK(status);
+
 #ifdef HAVE_MLX5_HW
     uct_ib_md_t *ib_md = (uct_ib_md_t *)md();
 
     if (amo_access) {
-        if (check_umr(ib_md)) {
+        if (ib_md->relaxed_order || check_umr(ib_md)) {
```
Comment (yosefe) [inline test/gtest/uct/ib/test_ib_md.cc:85]: unite the 2 if's:
if ((amo_access &umr) || relaxed_order ) {
  expect 2nd key
} else {
 expect only 1 key
}

**Pull request merged**